### PR TITLE
Update GrapplingHookListener

### LINKED ISSUE #2351: The grappling hook can be used as a leash
## :round_pushpin: Description (REQUIRED)

The grappling hook can be used as a leash when right click to entity and fence.

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

1. Give grappling hook.
2. Right click to entity and fence.

https://www.youtube.com/watch?v=wax_W7SmASM

## :bulb: Expected behavior (REQUIRED)

The grappling hook can not used as a leash

## :open_file_folder: /error-reports/ Folder

none

## :compass: Environment (REQUIRED)

 - Server Software (Spigot/Paper): Paper
 - Minecraft Version: 1.16.3
 - Slimefun Version: vDEV - 685 (git dc551907)
 - CS-CoreLib Version: vDEV - 91 (git e42a05d1)

### PR DESCRIPTION
**Pull request by andrefpoliveira**
## Description
Added a listener to the GrapplingHookListener to prevent users from leashing with the Grappling Hook.

## Changes
Added new function on GrapplingHookListener.java: onLeash()

## Related Issues
Resolves #2351 

## Checklist
- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java [hunk 1]
```diff
@@ -136,6 +137,16 @@ public void onPortalEnter(EntityPortalEnterEvent e) {
...
 
+    // Fixing Issue #2351
+    @EventHandler
+    public void onLeash(PlayerLeashEntityEvent e) {
+        if (!isEnabled()) {
+            return;
+        }
+
+        e.setCancelled(true);
+    }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:148]: Ehm, that just disables leashing as a whole...
So noone would be able to leash anything anymore.
You would still need to check if someone used a grappling hook.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java (hunk 1 above)
Comment (andrefpoliveira) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:148]: Oh okok sorry I thought that you had isolated the event! Let me try it then ;)

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:148]: Nope, event isolation sadly isn't a thing in the Bukkit API XD
You'll still need to add manual checks for things

Comment (andrefpoliveira) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:148]: @TheBusyBiscuit just a question, the CI is saying that could not found PlayerLeashEntityEvent should I use another event?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:148]: You imported it from the wrong namespaced, that's why.
It's located in the `org.bukkit.event.entity` package.
Source: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/PlayerLeashEntityEvent.html

**Comment by andrefpoliveira** [comment]
Sorry I havent testes the code because I'm not being able to transform it to jar. I hope that the check for the item in the Main Hand is enought to avoid colisions with other items

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java [hunk 2]
```diff
@@ -136,6 +139,23 @@ public void onPortalEnter(EntityPortalEnterEvent e) {
...
     }
 
+    // Fixing Issue #2351
+    @EventHandler
+    public void onLeash(PlayerLeashEntityEvent e) {
+        if (!isEnabled()) {
+            return;
+        }
+
+        Player p = (Player)e.getPlayer();
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:149]: No need to cast a Player to a Player here.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
One small redundancy, other than that this is looking good :+1:
Thanks again for making this pull request.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java (hunk 2 above)
Comment (andrefpoliveira) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java:149]: Oh right Sorry! It was the force of habbit from commands :) I will fix that and submit!

**Comment by TheBusyBiscuit** [comment]
> Sorry I havent tested the code because I'm not being able to transform it to jar. I hope that the check for the item in the Main Hand is enough to avoid collisions with other items
You can find short instructions on how to compile the project in our [CONTRIBUTING.md](https://github.com/Slimefun/Slimefun4/blob/master/CONTRIBUTING.md#toolbox-how-to-compile-slimefun4), you would just need to set it up as a Maven project first. There are plenty of online guides and tutorials too!

By the way it seems like your git account hasn't been configured correctly, as you are commiting from a local "BuildTools" instance and not from your account, check out this post for more info on how to set up your git username: https://docs.github.com/en/free-pro-team@latest/github/using-git/setting-your-username-in-git

**Review by TheBusyBiscuit — APPROVED** [review]

**Review by LinoxGH — APPROVED** [review]

**Pull request merged**